test/testdata/cfg/rescue_complex.rb

```diff
@@ -30,6 +35,44 @@ def multiple_rescue_classes()
     end
   end
 
+  def multiple_rescue_classes_varuse()
+    begin
+      meth
+    rescue LoadError, SocketError => baz
+      baz
+    end
+
+    T.reveal_type(baz) # error: Revealed type: `T.untyped`
```

Comment:
We want to be able to say `T.nilable(T.any(LoadError, SocketError))` here; #6928 gets us part of the way there, but to get further, we need to restructure exception handling a bit more.